code/datums/components/plumbing/IV_drip.dm

```diff
@@ -0,0 +1,21 @@
+///Component that feeds whatever it's supplied with to whoever is buckled to it. Can be safely added to whatever like the others, like chairs or something
+/datum/component/plumbing/IV_drip
+	demand_connects = SOUTH
+	supply_connects = NORTH
+
+	methods = INJECT
+
+/datum/component/plumbing/IV_drip/Initialize()
+	. = ..()
+
+	RegisterSignal(parent, list(COMSIG_IV_ATTACH), .proc/update_attached)
+	RegisterSignal(parent, list(COMSIG_IV_DETACH), .proc/clear_attached)
```

Comment:
Move these to a `RegisterWithParent`, then unregister then in `UnregisterWithParent`.